OrbitCore/LinuxEventTracer.h

```diff
@@ -5,46 +5,132 @@
 //-----------------------------------
 // Author: Florian Kuebler
 //-----------------------------------
-#pragma once
+
+#ifndef ORBIT_CORE_LINUX_EVENT_TRACER_H_
+#define ORBIT_CORE_LINUX_EVENT_TRACER_H_
 
 #include <atomic>
 #include <functional>
 #include <thread>
+#include <utility>
 
 #include "CallstackTypes.h"
 #include "LibunwindstackUnwinder.h"
+#include "OrbitFunction.h"
+#include "PrintVar.h"
+
+#if 0
+class LinuxFunction {
+ public:
+  LinuxFunction(std::string binary_path,
+                uint64_t file_offset,
+                uint64_t virtual_address)
+      : binary_path_{std::move(binary_path)},
+        file_offset_{file_offset},
+        virtual_address_{virtual_address} {}
+
+  const std::string& BinaryPath() const {
+    return binary_path_;
+  }
+
+  uint64_t FileOffset() const {
+    return file_offset_;
+  }
+
+  uint64_t VirtualAddress() const {
+    return virtual_address_;
+  }
+
+ private:
+  std::string binary_path_;
+  uint64_t file_offset_;
+  uint64_t virtual_address_;
+};
+#endif
+
+class LinuxEventTracerThread {
+ public:
+  LinuxEventTracerThread(pid_t pid, double frequency,
+                         std::vector<Function*> functions)
+      : pid_(pid), frequency_(frequency), functions_(std::move(functions)) {}
+
+  LinuxEventTracerThread(const LinuxEventTracerThread&) = delete;
+  LinuxEventTracerThread& operator=(const LinuxEventTracerThread&) = delete;
+  LinuxEventTracerThread(LinuxEventTracerThread&&) = default;
+  LinuxEventTracerThread& operator=(LinuxEventTracerThread&&) = default;
```

Comment:
also explicitly delete/default the default constructor?

Reply:
As discussed, we're not doing this as I'm already declaring a constructor.